src/stateReducer.js

```diff
@@ -0,0 +1,124 @@
+const ACTIONS = {
+  ADD_NEW_CART_ITEM: "add-new-cart-item",
+  BUY_NOW_HANDLER: "buy-now-handler",
+  UPDATE_WISHLIST: "update-wishlist",
+  INCREMENT_CART_COUNT: "increment-cart-count",
+  DECREMENT_CART_COUNT: "decrement-cart-count",
+  REMOVE_FROM_CART: "remove-from-cart",
+  REMOVE_FROM_WISHLIST: "remove-from-wishlist",
+  SORT: "Sort",
+  FILTER: "filter"
+};
+
```

Comment:
you can move the constants into a separate file & import them for implementation